src/components/form/FieldBase.vue

```diff
@@ -0,0 +1,258 @@
+<template>
+  <div
+    class="field"
+    :class="{
+      ['field--errored']: hasError,
+      ['field--active']: active,
+      ['field--filled']: filled,
+      ['field--minimized-label']: minimizedLabel,
+      ['field--disabled']: disabled,
+      ['field--grouped']: grouped,
+    }"
+  >
+    <div
+       v-if="icon"
+      class="field__icon"
+    >
+      <HdIcon
+        :src="icon"
+        role="presentation"
+      />
+    </div>
+    <div class="field__body">
+      <div class="field__main">
+        <slot v-bind="{ hasError }" />
+        <label
+          v-if="label"
+          :for="name"
+          class="field__label"
+        >
+          {{ label }}
+        </label>
+        <div class="field__input-right">
+          <slot name="input-right"/>
+        </div>
+        <div
+          v-if="!grouped"
+          class="field__border"
+        />
+      </div>
+      <p
```

Comment:
Do you think it makes sense to use a convention and add an `id` here and in the `label` based in the `name` of the input? 

This way when we are rendering an input, we can use `aria-labelledby` and `aria-describedby`.

``` html
<FieldBase
 :name="name"
  ...
>
	<input 
    type="text" 
    :name="name" 
    :id="name" 
    :aria-labelledby="`${name}-label`" 
    :aria-describedby="`${name}-helper`" 
  />
</FieldBase>

Reply:
The `<label>` element already has a `for="name"` and the input already has `id="name"`, so `aria-labelledby` is not necessary here. And the helper can't be as a description, because it's not consistent and it will be empty at first in most cases.
Or did I get you wrong?

Reply:
You are totally right =D
Thanks for the answer 😄